### ISSUE TITLE
Feature Request : Predefined Kit should allow to include accessories and consumables.

### ISSUE THREAD

**Issue by swift2512** [grokability/snipe-it#8762]
**Describe the solution you'd like**
Predefined Kit should allow to include accessories and consumables.

**Additional context**
In our company every new employee gets a pack of accessories: webcam, earphones and USB drive. This could easily be a KIT, but Snipe doesn't allow adding accessories or consumables to the KIT. Make kits for accessories and consumables. :)

IMHO, current implementation of KITs seems useless because I can't specify actual parts of this KIT. Basically, it pulls an available asset with lowest ID from specified model pool, but assets in kits should be strictly defined, not assigned randomly.

**Comment by gilesytheking** [comment]
+1 on this idea. I already had similar thoughts regarding the kits, along with some other users - #8723 #8539

**Comment by swift2512** [comment]
Just think about it for one moment.
Current process in steps:
1. You create a KIT in SnipeIT (PC, screen, printer) with only model numbers in it.
2. You assign this kit to a person.
3. SnipeIT assigns random assets to this person.
4. You check in SnipeIT which assets were assigned to the person.
5. You physically assemble this KIT by fetching all the assets from different locations.
 *Mindblown* Who came up with this idea? It's like putting cart before the horse. :)

The way it should be:
1. You physically assemble your kit and mark asset tags of used assets.
2. You create predefined kit in SnipeIT and state which assets it contains by putting asset tags from step#1.
3. You assign this KIT to a person.
4. SnipeIT assigns content of the KIT to this person.
*Less steps, less hassle, more control.*

**Comment by Bingmano** [comment]
I agree with the inclusion of accessories and consumables...I would also expand to components and possibly licenses as well.  When issuing a new device to someone, I include a case (accessory) and often times include components (dock, pen, mouse).  As for licenses, it would be beneficial as when a user needs extra components, it is often because they have specialized software as well.

**Comment by snipe** [comment]
Components are things like hard drives , RAM, etc, and are meant to be assigned to assets. Dock, pen, mouse are accessories, and meant to be assigned to users.

**Comment by Bingmano** [comment]
Well in my case, I leverage components for surface type covers.  they are assigned to a specific asset and when they are returned (depending on their condition) are reused or recycled and replaced.

**Comment by gilesytheking** [comment]
Snipe has become a vital tool for many use cases. Adding additional options to suit various uses can only be a step forward. Perhaps having the option to enable/disable certain functionality within the global settings would give more flexibility to admins to  setup Snipe-IT for their use-case.

For my use case the kits should definitely include Accessories & licences, but this may not apply to everyone. Having the ability to add specific assets into a kit as apposed to randomly selecting one, would be preferred. However random selection of accessories & licences would work for my setup, as they are just quantity based items.

**Comment by StoneCotton** [comment]
Coming back to this from another perspective, I just started using Snipe-It for inventory management in my video production company. So far it's been working great, but adding this functionality to the kits function would be great. We have kits which include cameras, filters, and batteries. While cameras and lenses are assets, things like SD cards and batteries are more accessories which come packaged in a premade kit. Being able to just select that kit as being checked out feels a lot smoother compared to having to go in and select them all manually every time.

**Comment by NMX-Drew** [comment]
I'd also like to +1 this request. Would be great for our use cases where we need to include accessories (GPS antenna, power supply, ethernet cables, ect).

**Comment by G13312** [comment]
Yes please that would be great!! Would give us the ability to bundle up all the accessories/consumables required for specific assets!

**Comment by henningkessler** [comment]
I am joining the request as well ;-)

**Comment by Lari0** [comment]
Joining this one too .

**Comment by tweekmonster** [comment]
Adding to the request as well. If the blocker for this is that shoehorning consumables in the existing code is a hassle, could there perhaps be a new feature called "bundles"? Basically, kits that we don't expect to be returned.

In our org, we send out a swag kit/bundle to new hires (several t-shirts, magnet, mouse pad, etc). It would be helpful if the people involved are able to auto-checkout the correct amount of consumables when assembling a bundle for shipping. Not to mention seeing that there is a supply shortage that would make the bundles incomplete. While we could pre-assemble the bundles and make those a consumable on their own, it wouldn't work out very well since we send out individual parts of the bundles upon request.

**Comment by snipe** [comment]
The blocker is more of a UI issue than anything else. The original PR was a little broken, but it was also really unpleasant and confusing to use. We just need to find the time to sort it out.

**Comment by tweekmonster** [comment]
@snipe Thanks for the update.  I can relate on UI being a hassle.  If it makes a difference in prioritization, we did sign up for the monthly hosted plan a couple hours before I posted my message 😬🙏

In any case, Snipe-IT is a great product as it is now.  Our operations manager is genuinely psyched about not using spreadsheets to manage our inventory.

**Comment by snipe** [comment]
@tweekmonster tysm for the kind words! We’re definitely planning on doing a lot of work on that coming up :) (Also a buttload of other UX/UI/workflow improvements). 

We’re hoping to have Snipe-IT V6 out by oct of this year, so many many changes are coming!

**Comment by SecurusCmi** [comment]
Late to the party but jumping on the bandwagon too! 

An extension of the predefined Kit would be amazing (and make my job a lot easier).

Have been using Snipe for around 18 months+ and I am blown away by the amount of positive development work done to improve this product. I can't wait to see what the next big update brings!! 

**Keep up the good work!**

**Comment by ByJacob** [comment]
Any progress on this matter?

**Comment by kamsobe** [comment]
Version 6 is out, but still no sign of expanded Predefined Kit functionality? Anyone have a workaround?

We need the low inventory reminders that Accessories offer, but we also need the ability to assign multiple accessories to a user at once... We're not yet tracking accessories down to the S/N, just making sure everyone has 1 of each item...

**Comment by TraumaSK** [comment]
Also adding a +1 to this request for the same reasons already mentioned.  

Just started using the app, and I really enjoy it so far!

**Comment by mine-ing** [comment]
May I also join the bandwagon with #10986 considering how much easier it would make life

**Comment by eholloway135** [comment]
It would be nice if kits, when assigned could go off of what's available to specific locations, rather than just pulling from the total of what's available. 
Manage five locations and from testing, it'll pull from whatever is available no matter the product's default locations or the product or location of the user being assigned the kit.

**Comment by cameronaffleck** [comment]
It would be great to be able to assign specific assets to the kits, as I don't want it to choose a random asset. My kits live all together already and therefore want to be able to keep it all together.

**Comment by bk-lg** [comment]
This would be a great feature for providing laptops bundled with the correct chargers/power supplies.

**Comment by swift2512** [comment]
It's just sad that after 3 years and 10 new versions, snipeit still uses same broken model for predefined kits. Accessories can't be checked out in bulk or to anything else other than user.

**Comment by snipe** [comment]
@swift2512 PRs are always welcome. Passive-aggressive comments like this are not.

**Comment by mgtmgtmgt** [comment]
Is there a solution to this? Need to be able to add peripherals and licenses to kits.

**Comment by myates1** [comment]
New to Snipe-it just setting up, but I have to agree adding accessories and licenses to a package would be a very useful feature

**Comment by BJLawsonSTM** [comment]
This would be a great feature. I work for a school district, and we are trying to have a kit that we can check out that includes things such as a Promethean remote, pens, and presentation clickers. It would be awesome to be able to check out one thing for a teacher that includes items that don't have unique serials.

**Comment by biasedlogic** [comment]
Got here by googling predefined kits. I am also puzzled by the present functionality, or rather the lack of.

My situation:
I run robotic classes. For the robotic classes I use LEGO robotic sets. Each set is comprised of two boxes, say, Box A and Box B. One of the boxes contains, beyond other things, several electronic components which need to be tracked individually. 

The boxes are either 
a) checked out as a pair (a fixed pair, always A1+B1 or A5+B5, never mixed like A3+B2), or 
b) only Box A (the one with electronics), or 
c) just some of the electronic components individually.

I kinda thought I could define fixed sets, which would automagically check if all assets assigned to this set are available for check-out, so that I could just select 'I want one full robotics kit' and it would tell me that the component set number 3, 5 and 7 are all available (say, because Box a from set 1 is checked out and because some components from sets 2 and 6 are checked out).

For the moment being I have a workaround where everything is an asset, but all electronic elements are checked out to an asset being the box A. If someone checks out just a few electronic components, the box is marked non-deployable and its assets get checked in + out to specific person, but it's kinda cumbersome.

Is there a solution for this kind of a setup?

**Comment by Naddox** [comment]
@snipe Any progress on this feature request?

**Comment by Batch80** [comment]
another +1

Kit bundle for new starters = 1 laptop (Asset), + Bag (Accessory) + Headset (Accessory) + Maybe Monitor (asset) but we'd need to be able to identify which assets we wanted to issue, not just be assigned one from the pool.

**Comment by dreamer2038** [comment]
I also need this functionality. It would be better if you can customize the fields.
Because the operation process here is relatively smooth, otherwise you need to operate on 3 interfaces to achieve the goal.

Because receiving accessories or consumables is about the person, not the device.
Maintenance is limited to related equipment

**Comment by nathanelijahtorres** [comment]
We're currently in the middle of revamping our AMS and we would really like this feature. @snipe 

Thank you for your efforts !

**Comment by tiago-matos-jta** [comment]
@snipe Any news on this?

We give our employees Welcome kits (contains an agenda, pens, a coat, shirt and hoodie, etc), and we'd like to check out a kit and deduct the defined quantity defined on the kit out of the available quantity of said consumables.

**Comment by snipe** [comment]
@tiago-matos-jta The backend already largely supports it, but we had to pull a lot of the front end since it was a mess and the UX was baffling.

If you'd like to work on this to contribute back to Snipe-IT, please talk to us so we can work through what the feature should actually do, how it should behave on the front-end, etc. We've gotten quite a few use-cases over time.

**Comment by tiago-matos-jta** [comment]
@snipe Forgot to reply - so I'm going to host the platform on a sandbox so I can mess around with this, I want to see if I'm able to code it properly!

**Comment by JoshJames04** [comment]
Like many others I think this feature would change a lot for my use case and would be greatly beneficial.

**Comment by HowDoIEvenIT** [comment]
Yet another request for this feature please.

**Comment by AB-131015** [comment]
I think this would be very useful

**Comment by katosan** [comment]
I would second  that, i want to use snipe-it at my school  and as we  are handing (and retrieving) ~200 iPad Kits  on  a regular basis that would help immensly

**Comment by danbyangasga** [comment]
Requesting to add licenses, accessories and consumables on this

**Comment by Agwstin** [comment]
+1

**Comment by Marlon-Wessel** [comment]
I would like to second this. When we're onboarding new employees, they need a laptop, two monitors a docking station, a webcam, a headset and an MnK set. Laptops, monitors and docking stations are assets while headsets, webcams and MnK sets are accessories as intended I'd like to think. Why can I not add them to one predefined kit to deploy all needed hardware with a single click?
